rgw/v2/tests/s3_swift/reusables/server_side_encryption_s3.py

```diff
@@ -44,27 +46,37 @@ def get_object_encryption(s3_client, bucketname, s3_object_name):
     """
     get object encryption for an object, it should be AES256
     """
-    log.info(f"Test if object uploaded is encrypted with AES256")
     get_obj_encryption = s3_client.get_object(Bucket=bucketname, Key=s3_object_name)
-    log.info(
-        f"server side encryption enabled on {s3_object_name} is : {get_obj_encryption['ServerSideEncryption']}"
-    )
     result = get_obj_encryption["ServerSideEncryption"]
-    if result != "AES256":
+    if result == "AES256":
+        log.info(f"server side encryption is with s3 keys on object {s3_object_name}")
+    elif result == "aws:kms":
+        log.info(f"server side encryption is with kmsi keys on object {s3_object_name}")
+    else:
         raise TestExecError("object encryption has failed")
 
 
-def put_object_encryption(s3_client, bucketname, s3_object_name):
+def put_object_encryption(s3_client, bucketname, s3_object_name, encryption_method):
     """
     put object encryption for an object, it should be AES256
     """
     log.info(f"Enable object encryption when object is uploaded.")
-    put_obj_encryption = s3_client.put_object(
-        Bucket=bucketname,
-        Key=s3_object_name,
-        Body="Sinister sisters hide in plain view",
-        ServerSideEncryption="AES256",
-    )
+    if encryption_method == "s3":
+        log.info("server side encryption is with s3 keys")
+        put_obj_encryption = s3_client.put_object(
+            Bucket=bucketname,
+            Key=s3_object_name,
+            Body="Sinister sisters hide in plain view",
```

Comment:
@viduship Could you please confirm that are we uploading same content for n times

Reply:
@udaysk23 , it is addressed now